### PR TITLE
feat: add Reddit social icon

### PR DESCRIPTION
**Pull request by mehalter**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content
- Changes to Starlight code

#### Description

This adds Reddit as a social icons for projects that want to be able to add links to their official subreddit.

> I am taking part in Hacktoberfest 😄

### CODE REVIEW

**Review by HiDeoo — COMMENTED** [review]
Thanks for your contribution :raised_hands: 

- This change will require a changeset like [this one](https://raw.githubusercontent.com/withastro/starlight/56e311d616f1f72d5ca8f3f2fb192b36c11f3e34/.changeset/slimy-books-travel.md) to inform our users about the new icon (you can generate one using the `pnpm exec changeset` command)
- Would you mind reverting the changes to non-English doc pages that were introduced and only keep the changes for the English version? We usually only update the English pages and leave changes to the translations to our translators as this makes it easier for the [Starlight translation tracker](https://i18n.starlight.astro.build/) to keep track of the changes.

**Comment by mehalter** [comment]
Sure no problem @HiDeoo ! I have updated my commit to add the changeset and remove the international updates.

**Review by HiDeoo — APPROVED** [review]
Thanks for the update, this looks good to me 🙌

**Review by delucis — COMMENTED** [review]
Thanks for adding this @mehalter! Added one suggestion but other than that, this looks great and we can get it out in our next patch release.

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -95,4 +95,6 @@ export const Icons = {
...
...1054,5.11054,0,0,0,12.042,19.5a5.12,5.12,0,0,0,3.78223-1.2168,1.00058,1.00058,0,1,0-1.41406-1.416ZM9.2005,15.00165a1,1,0,1,0-1.0003-.9997A1.00079,1.00079,0,0,0,9.2005,15.00165Zm6-2a1,1,0,1,0,.9997,1.0003A1.00138,1.00138,0,0,0,15.2005,13.00165ZM23,11.78027a3.77157,3.77157,0,0,0-6.794-2.26471,16.50461,16.50461,0,0,0-3.05005-.47851l.85578-5.705,2.08752.70984a2.99694,2.99694,0,0,0,5.99353-.06433V3.95508a3.02886,3.02886,0,0,0-3-2.95508,2.97689,2.97689,0,0,0-2.33209,1.155L13.52246,1.05371a.999.999,0,0,0-1.31152.79785L11.13446,9.027A16.66426,16.66426,0,0,0,7.794,9.51556a3.76753,3.76753,0,0,0-6.22492,4.23487A4.86206,4.86206,0,0,0,1,16c0,3.9248,4.832,7,11,7s11-3.0752,11-7a4.86217,4.86217,0,0,0-.56866-2.2489A3.78344,3.78344,0,0,0,23,11.78027ZM19.09277,3a1,1,0,1,1-1,1A1.01672,1.01672,0,0,1,19.09277,3ZM4.78027,10a1.75976,1.75976,0,0,1,.88172.24951A9.97889,9.97889,0,0,0,3.0141,11.9234c-.004-.04785-.0141-.095-.0141-.14313A1.78255,1.78255,0,0,1,4.78027,10ZM12,21c-4.87891,0-9-2.29-9-5s4.12109-5,9-5,9,2.29,9,5S16.87891,21,12,21Zm8.9859-9.07654A9.97805,9.97805,0,0,0,18.338,10.24951,1.75993,1.75993,0,0,1,19.21973,10,1.78255,1.78255,0,0,1,21,11.78027C21,11.82837,20.98993,11.87561,20.9859,11.92346Z"/>'
```
[the remark this answers is not in the document]
Comment (mehalter) [reply packages/starlight/components/Icons.ts]: Thanks for the suggestion! This has been resolved in the latest commit

**Review by delucis — APPROVED** [review]
Thanks! Should be good to go for our next release 🙌 

We’ll merge it shortly before release to avoid updating the docs site until the Reddit icon is available.

**Comment by mehalter** [comment]
Awesome sounds great! @delucis I also went ahead and looked at optimizing the rest of the UI icons in that file as some of them are much larger than they need to be. I went ahead and opened another PR. Let me know if this is something you are interested in or if you would prefer it be in this PR as well.

**Comment by delucis** [comment]
Nice that’s very helpful — thanks @mehalter. A separate PR makes sense for that I think 🙌

**Comment by mehalter** [comment]
With that other large change to the `Icons.ts` file git wasn't smart enough to resolve the diff so I went ahead and did a rebase on the current `main`.

Thanks for letting me make some small contributions. Great job on this project, it is awesome!

**Review by delucis — APPROVED** [review]
Thanks for resolving those — looks good!

**Pull request merged**